test/jdk/java/lang/Thread/virtual/ThreadAPI.java

```diff
@@ -63,31 +63,28 @@
 import java.nio.channels.Selector;
 
 import jdk.test.lib.thread.VThreadRunner;
-import org.testng.SkipException;
-import org.testng.annotations.AfterClass;
-import org.testng.annotations.BeforeClass;
-import org.testng.annotations.DataProvider;
-import org.testng.annotations.Test;
-import static org.testng.Assert.*;
-
-public class ThreadAPI {
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assumptions.*;
+
+class ThreadAPI {
     private static final Object lock = new Object();
 
     // used for scheduling thread interrupt
-    private ScheduledExecutorService scheduler;
-
-    @BeforeClass
-    public void setUp() throws Exception {
-        ThreadFactory factory = (task) -> {
-            Thread thread = new Thread(task);
-            thread.setDaemon(true);
-            return thread;
-        };
+    private static ScheduledExecutorService scheduler;
+
+    @BeforeAll
+    static void setup() throws Exception {
+        ThreadFactory factory = Executors.defaultThreadFactory();
```

Comment:
Unlike previously, this change now uses a `ThreadFactory` which creates non-daemon threads. Is that intentional?

Reply:
It's not an issue. The `@AfterAll` finish method does the shutdown so it doesn't matter if the STPE threads are daemon or not.